R/textstat_simil.R

```diff
@@ -105,22 +123,48 @@ textstat_simil.dfm <- function(x, selection = NULL,
             stop(paste(selection[is.na(i)], collapse = ", "), " does not exist")
     }
     if (margin == "features") {
-        result <- textstat_proxy(x, x[, i], margin, method, 1, use_na = TRUE)
+        temp <- textstat_proxy(x, x[, i], margin, method, 1, 
+                               min_proxy = if (min_simil == 0) NULL else min_simil, use_na = TRUE)
+                               
     } else {
-        result <- textstat_proxy(x, x[i, ], margin, method, 1, use_na = TRUE)
+        temp <- textstat_proxy(x, x[i, ], margin, method, 1, 
+                               min_proxy = if (min_simil == 0) NULL else min_simil, use_na = TRUE)
     }
-    result <- as_dist(result, method, match.call(), diag = diag, upper = upper)
-    if (is.null(selection)) {
-        class(result) <- c("simil", "dist")
-    } else {
-        class(result) <- c("simil_selection", "dist_selection")
+    
+    if (upper) temp <- as(temp, "dgTMatrix")
+    
+    # ensure original sort order
+    xnames <- if (margin == "documents") docnames(x) else featnames(x)
+    ynames <- if (is.null(selection)) xnames else xnames[i]
+    temp <- temp[xnames, ynames, drop = FALSE]
+
+    result <- data.frame(x = factor(temp@i + 1L, seq_len(nrow(temp)), rownames(temp)),
+                         y = factor(temp@j + 1L, seq_len(ncol(temp)), colnames(temp)),
+                         similarity = temp@x)
+    
+    # eliminate identical pairs when diag = FALSE
+    if (!diag) {
+        result <- result[result[1] != result[2], ]
     }
-    result
+    
+    # replace x and y with margin names
+    names(result)[1:2] <- paste0(stri_sub(margin, 1, -2), 1:2)
+    
+    class(result) <- c("textstat_simil", "data.frame")
```

Comment:
Output from `textstat_collocations` is `c('collocations', 'textstat', 'data.frame')`, so this should be `c('simil', 'textstat', 'data.frame').

Reply:
See #1690. Classing these as `simil` and `dist` would collide with return values from `stats::dist()` and `proxy::dist()`.

The subsetting seems to preserve the attributes fine. And conveniently, it does not print them.
```r
> dfmat <- dfm(corpus_subset(data_corpus_inaugural, Year > 2000), 
+              remove_punct = TRUE, remove = stopwords("english"))
> str(textstat_simil(dfmat, method = "cosine", margin = "documents")[1:3, ])
Classes ‘textstat_simil’ and 'data.frame':	3 obs. of  3 variables:
 $ document1 : Factor w/ 5 levels "2001-Bush","2005-Bush",..: 1 2 3
 $ document2 : Factor w/ 5 levels "2001-Bush","2005-Bush",..: 1 1 1
 $ similarity: num  1 0.52 0.541
 - attr(*, "method")= chr "cosine"
 - attr(*, "min_simil")= num 0
 - attr(*, "diag")= logi TRUE
 - attr(*, "upper")= logi TRUE
> textstat_simil(dfmat, method = "cosine", margin = "documents")[1:3, ]
   document1 document2 similarity
1  2001-Bush 2001-Bush  1.0000000
2  2005-Bush 2001-Bush  0.5204355
3 2009-Obama 2001-Bush  0.5411649
```